src/super_gradients/training/models/model_factory.py

```diff
@@ -189,32 +189,35 @@ def get_model_name(model: torch.nn.Module) -> Optional[str]:
-    :param pretrained_weights:  Describe the dataset of the pretrained weights (for example "imagenent").
-    :param load_backbone:       Load the provided checkpoint to model.backbone instead of model.
-    :param download_required_code: if model is not found in SG and is downloaded from a remote client, overriding this parameter with False
-                                    will prevent additional code from being downloaded. This affects only models from remote client.
+    :param model_name:              Defines the model's architecture from models/ALL_ARCHITECTURES
+    :param arch_params:             Architecture hyper parameters. e.g.: block, num_blocks, etc.
+    :param num_classes:             Number of classes (defines the net's structure).
+                                        If None is given, will try to derive from pretrained_weight's corresponding dataset.
+    :param strict_load:             See super_gradients.common.data_types.enum.strict_load.StrictLoad class documentation for details
+                                        (default=NO_KEY_MATCHING to suport SG trained checkpoints)
+    :param checkpoint_path:         The path to the external checkpoint to be loaded. Can be absolute or relative (ie: path/to/checkpoint.pth) path or URL.
+                                        If provided, will automatically attempt to load the checkpoint.
+    :param pretrained_weights:      Describe the dataset of the pretrained weights (for example "imagenent").
+    :param load_backbone:           Load the provided checkpoint to model.backbone instead of model.
+    :param download_required_code:  If model is not found in SG and is downloaded from a remote client, overriding this parameter with False
+                                        will prevent additional code from being downloaded. This affects only models from remote client.
     :param checkpoint_num_classes:  num_classes of checkpoint_path/ pretrained_weights, when checkpoint_path is not None.
-     Used when num_classes != checkpoint_num_class. In this case, the module will be initialized with checkpoint_num_class, then weights will be loaded. Finaly
-        replace_head(new_num_classes=num_classes) is called (useful when wanting to perform transfer learning, from a checkpoint outside of
-         then ones offered in SG model zoo).
-
+                                        Used when num_classes != checkpoint_num_class. In this case, the module will be initialized with checkpoint_num_class,
+                                        then weights will be loaded.
+                                        Finaly replace_head(new_num_classes=num_classes) is called (useful when wanting to perform transfer learning,
+                                        from a checkpoint outside of then ones offered in SG model zoo).
+    :param num_input_channels:      Number of input channels.
```

Comment:
I beleive this parameter should be introduced in checkpoint params and forwarded to models.get every time we call models.get (Multiple places in Trainer, train_from_recipe etc)

Reply:
Added here [9d86c3e](https://github.com/Deci-AI/super-gradients/pull/1557/commits/9d86c3edded9ac584764f8e8250b93f839de68f0) [0a38118](https://github.com/Deci-AI/super-gradients/pull/1557/commits/0a381185d2335b97c0bd0f800032840d4467ca61) [a5e3c86](https://github.com/Deci-AI/super-gradients/pull/1557/commits/a5e3c86b455d7227c1fe38b861afc0a6819c1929)